jobs/build/send-umb-messages/Jenkinsfile

```diff
@@ -25,33 +25,76 @@ node {
                     returnStdout: true,
                     script: "curl -s https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/${release}/latest",
                 ).trim()
-		try {
+                try {
                     previousRelease = readFile("${release}.current")
-		} catch (readex) {
-		    // The first time this job is ran and the first
-		    // time any new release is added the 'readFile'
-		    // won't find the file and will raise a
-		    // NoSuchFileException exception.
-		    touch file: "${release}.current"
-		    previousRelease = ""
-		}
+                } catch (readex) {
+                    // The first time this job is ran and the first
+                    // time any new release is added the 'readFile'
+                    // won't find the file and will raise a
+                    // NoSuchFileException exception.
+                    touch file: "${release}.current"
+                    previousRelease = ""
+                }
 
                 if ( latestRelease != previousRelease ) {
-		    currentBuild.displayName += "🆕: ${release} "
-		    currentBuild.description += "\n🆕: ${release}"
+                    currentBuild.displayName += "🆕: ${release} "
+                    currentBuild.description += "\n🆕: ${release}"
+
+                    def object = readJSON text: latestRelease
+
+                    def downloadURL = object.downloadURL
+                    def version = object.name
+                    def pullRef = object.pullSpec
+                    def digest = "MISSING_DIGEST_FIX_PLEASE"
```

Comment:
I wasn't sure how to get the digest myself. I opened a PR and got most of the work done

https://github.com/openshift/release-controller/pull/92

Not sure how to wrap it up though. Hey, @smarterclayton or @stevekuznetsov any thoughts on how to wrap that up? I just want to include the digest of the release in the stream API output, like in here: https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/4.1.0-0.ci/latest

Reply:
Hello, it seems a bit stuck...can I help somehow?

Reply:
It seems like Tim got directions in that PR but has been working on higher priorities since. Maybe something @vfreex and/or @shiywang could move forward.

Although, why do we need the release-controller to expose the digest? Can't we just `skopeo inspect` the pullspec?

Reply:
@sosiouxme  I suppose that would work too :smiley_cat:

Reply:
We need the digest because of [Factory 2.0 standard](https://pagure.io/fedora-ci/messages/blob/master/f/schemas/container-image.yaml) -  it's a required attriubute

Reply:
Sorry for my late reply. I think `skopeo inspect` should work, but returning a digest in release-controller seems to be more robust way.

Actually I was one of major guys who drafted the `container-image` artifact format in that standard :)

Reply:
For this job to move forward here is what we need

`<braindump>`
* Make a new function in commonlib or something
* that function runs the `oc image info` business
* It returns json

```groovy
// pipeline-scripts/commonlib.groovy

def image_info_json(String pullSpec) {
    def result = commonlib.shell(
        script: "oc image info -o json ${pullSpec}"
        returnAll: true,
    )

    if ( result.returnStatus != 0) {
         // intelligent error condition handling
    }

    readJSON(text: result.stdout)
}
```

Take that result and directly reference the `digest` attribute

```groovy

    def digest = commonlib.image_info_json(object.pullSpec).digest
```

`</braindump>`

Reply:
@tbielawa @vfreex agreed, use `oc image info -o json` to look up the info.